src/System.Private.CoreLib/shared/System/SpanHelpers.Byte.cs

```diff
@@ -161,28 +172,135 @@ public static unsafe bool Contains(ref byte searchSpace, byte value, int length)
                 offset += 1;
             }
 
-            if (Vector.IsHardwareAccelerated && ((int)(byte*)offset < length))
+            // We get past SequentialScan only if IsHardwareAccelerated or intrinsic .IsSupported is true; and remain length is greater than Vector length.
+            // However, we still have the redundant check to allow the JIT to see that the code is unreachable and eliminate it when the platform does not
+            // have hardware accelerated. After processing Vector lengths we return to SequentialScan to finish any remaining.
+            if (Avx2.IsSupported)
             {
-                lengthToExamine = (IntPtr)((length - (int)(byte*)offset) & ~(Vector<byte>.Count - 1));
+                if ((int)(byte*)offset < length)
+                {
+                    if ((((nuint)Unsafe.AsPointer(ref searchSpace) + (nuint)offset) & (nuint)(Vector256<byte>.Count - 1)) != 0)
```

Comment:
This `if` chunk isn't needed, its only `.IndexOf` that needs the strict alignment

Reply:
Thanks, I removed the check. Updated perf numbers for bytes:

``` ini

BenchmarkDotNet=v0.11.5, OS=Windows 10.0.17134.706 (1803/April2018Update/Redstone4)
Intel Core i7-8700K CPU 3.70GHz (Coffee Lake), 1 CPU, 12 logical and 6 physical cores
Frequency=3609378 Hz, Resolution=277.0560 ns, Timer=TSC
.NET Core SDK=3.0.100-preview5-011281
  [Host]     : .NET Core 3.0.0-preview5-27613-06 (CoreCLR 4.6.27613.71, CoreFX 4.700.19.21214), 64bit RyuJIT
  Job-WKQEKY : .NET Core ? (CoreCLR 4.6.27704.0, CoreFX 4.700.19.21911), 64bit RyuJIT
  Job-EZEXQF : .NET Core ? (CoreCLR 4.6.27704.0, CoreFX 4.700.19.21911), 64bit RyuJIT


```
|        Method | Toolchain |     Mean |     Error |    StdDev |   Median |      Min |      Max | Ratio |
|-------------- |---------- |---------:|----------:|----------:|---------:|---------:|---------:|------:|
|  ContainsTrue |    Master | 2.235 us | 0.0065 us | 0.0055 us | 2.233 us | 2.229 us | 2.250 us |  1.00 |
|  ContainsTrue |      Test | 1.076 us | 0.0017 us | 0.0015 us | 1.076 us | 1.073 us | 1.079 us |  0.48 |
|               |           |          |           |           |          |          |          |       |
| ContainsFalse |    Master | 3.195 us | 0.0128 us | 0.0120 us | 3.191 us | 3.181 us | 3.216 us |  1.00 |
| ContainsFalse |      Test | 2.914 us | 0.0062 us | 0.0052 us | 2.914 us | 2.905 us | 2.924 us |  0.91 |